src/renderer/atlas/atlas.vcxproj

```diff
@@ -0,0 +1,55 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project DefaultTargets="Build" ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup>
+    <ProjectGuid>{8222900C-8B6C-452A-91AC-BE95DB04B95F}</ProjectGuid>
+    <Keyword>Win32Proj</Keyword>
+    <RootNamespace>atlas</RootNamespace>
+    <ProjectName>RendererAtlas</ProjectName>
+    <TargetName>ConRenderAtlas</TargetName>
+    <ConfigurationType>StaticLibrary</ConfigurationType>
+  </PropertyGroup>
+  <Import Project="$(SolutionDir)src\common.build.pre.props" />
+  <ItemGroup>
+    <ClCompile Include="AtlasEngine.api.cpp" />
+    <ClCompile Include="AtlasEngine.r.cpp" />
+    <ClCompile Include="pch.cpp">
+      <PrecompiledHeader>Create</PrecompiledHeader>
+    </ClCompile>
+    <ClCompile Include="AtlasEngine.cpp" />
+  </ItemGroup>
+  <ItemGroup>
+    <ClInclude Include="pch.h" />
+    <ClInclude Include="AtlasEngine.h" />
+  </ItemGroup>
+  <ItemGroup>
+    <FxCompile Include="shader_ps.hlsl">
+      <ShaderType>Pixel</ShaderType>
+      <ShaderModel>4.1</ShaderModel>
+      <AllResourcesBound>true</AllResourcesBound>
+      <VariableName>shader_ps</VariableName>
+      <ObjectFileOutput />
+      <HeaderFileOutput>$(OutDir)$(ProjectName)\%(Filename).h</HeaderFileOutput>
+      <TreatWarningAsError>true</TreatWarningAsError>
+      <AdditionalOptions>/Zpc %(AdditionalOptions)</AdditionalOptions>
+      <AdditionalOptions Condition="'$(Configuration)'=='Release'">/O3 /Qstrip_debug /Qstrip_reflect %(AdditionalOptions)</AdditionalOptions>

```

Comment:
magic compiler flags -- cause for concern?

Reply:
They're not that magic at all! For some weird reason neither of these 3 parameters are available as msbuild options.
But they're what we at Microsoft use ourselves in our own sample code. If you search for these inside this GitHub org you'll find dozens if not hundreds of results. 🙂
`/O3` is for future compat and enabled all available optimizations and `/Qstrip_debug/reflect` remove debug/reflection metadata from the bytecode, reducing it's size _a lot_.